usr/src/uts/common/fs/zfs/vdev_disk.c

```diff
@@ -37,6 +37,13 @@
 #include <sys/efi_partition.h>
 #include <sys/fm/fs/zfs.h>
 
+/*
+ * Tunable parameter for debugging or performance analysis.  Setting
+ * zfs_nocacheflush will cause corruption on power loss if a volatile
+ * out-of-order write cache is enabled.
+ */
+boolean_t zfs_nocacheflush = B_FALSE;
```

Comment:
You're suggesting that we add a similar condition statement in `vdev_file_io_start`, to avoid calling `VOP_FSYNC` when this is set?

I'm not opposed to that, but I'd rather do that in a separate change, since that'd change the behavior of this existing `zfs_nocacheflush` tunable.

i.e. with my changes, if this tunable is already set (e.g. on a current system), the behavior will mostly stay the same (still, no flushes set from the ZIL or SPA) after upgrading to this change (w.r.t. the tunable, at least).. whereas what (I think) you're suggesting would cause different behavior on that system with the tunable set, after upgrading to this change.

What do you think? Would you be OK leaving that for a follow-up change?

Reply:
Yeah, totally fine for a follow-up.  I imagine it's an extremely small population of folks that might be disabling cache flushes and also using vdev_file.